File: proj01/ruby/rectangle_area.rb
#! /usr/bin/ruby
# rectangle_area.rb computes the area of a rectangle given a width and length
#
# Input: the width and length of a rectangle
# Precondition: the width and length must be positive
# Output: the area of the rectangle
#
# Author: David Reidsma
# Date: 11/02/2020
################################################################

# function rectangleArea returns a rectangle's area given width and length
# Parameter: width, a number; must be greater than 0
# Parameter: length, a number; must be greater than 0
# Returns: the area of rectangle whose sides are of width and length
def rectangleArea(width, length)
  width * length
end

if __FILE__ == $0
  puts "\nTo compute the area of a rectangle,"
  
  print " enter its width: "
  width = gets.chomp.to_f

  print " enter its length: "
  length = gets.chomp.to_f

  print "\nThe area is: "
  area = rectangleArea(width, length)
  puts area
end

File: proj01/java/RectangleArea.java
/**
 * RectangleArea.java computes the area of a rectangle.
 *
 * Input: The width and length of a rectangle.
 * Precondition: The width and length are positive numbers.
 * Output: The area of the rectangle.
 *
 * Author: David Reidsma
 * Date: 11/02/2020
 **********************************************************/

import java.util.Scanner;

public class RectangleArea {
	/**
	 * Returns the area of a rectangle.
	 * @param width	The width of the rectangle. Must be positive.
	 * @param length The length of the rectangle. Must be greater than one.
	 * @return The area of the rectangle.
	 */
	private static double rectangleArea(double width, double length) {
		return width * length;
	}

	public static void main(String[] args) {
		System.out.println("\nTo compute the area of a rectangle,");
		
		Scanner scanner = new Scanner(System.in);
		
		System.out.print(" enter a width: ");
		double width = scanner.nextDouble();

		System.out.print(" enter a length: ");
		double length = scanner.nextDouble();

		double area = rectangleArea(width, length);
		System.out.println("\nThe area is " + area);
	}
}
